### PR TITLE
Update app-launch topic

### PR DESCRIPTION
**Pull request by richlander**
An attempt to improve the flow and effectiveness of the doc for a variety of users.

Related to #28805.

@elinor-fung @vitek-karas

[Current topic](https://docs.microsoft.com/dotnet/core/runtime-discovery/troubleshoot-app-launch) 
[Internal-only rendered view](https://review.docs.microsoft.com/en-us/dotnet/core/runtime-discovery/troubleshoot-app-launch?branch=pr-en-us-29738)

### CODE REVIEW

**Review by tdykstra — APPROVED** [review]
LGTM, a few suggestions

File: docs/core/runtime-discovery/troubleshoot-app-launch.md [hunk 1]
```diff
@@ -102,21 +110,42 @@ The following frameworks were found:
 
 The error indicates the name, version, and architecture of the missing framework and the location at which it is expected to be installed. To run the application, you can [install a compatible framework](#install-a-compatible-framework) at the specified ".NET location". If the application is targeting a lower version than one you have installed and you would like to run it on a higher version, you can also [configure roll-forward behavior](#configure-roll-forward-behavior) for the application.
 
-### Install a compatible framework
+### Install a compatible runtime
+
+The error message includes a link to download the missing framework. You can follow this link to get to the appropriate download page.
```
Comment (vitek-karas) [inline docs/core/runtime-discovery/troubleshoot-app-launch.md:115]: Do we use "runtime" or "framework"?

You changed the header to "runtime", but that use "framework".

I don't know which one we prefer - but it seems to be "runtime" for all external facing documents/web sites (internally it's called framework, but that should not be a problem).

Comment (elinor-fung) [reply docs/core/runtime-discovery/troubleshoot-app-launch.md:115]: The host error message itself does use 'framework' though. The way I read the updates in this section is 'runtime' refers to the download from the .NET downloads page and a runtime download contains one or more 'framework's. Overloaded terms for sure, but I think that does at least match our [glossary](https://docs.microsoft.com/dotnet/standard/glossary).

Comment (vitek-karas) [reply docs/core/runtime-discovery/troubleshoot-app-launch.md:115]: Good - at least we're consistent. Maybe a link to the glossary might be worth it? (I didn't know we had that page, it's pretty good)

Comment (richlander) [reply docs/core/runtime-discovery/troubleshoot-app-launch.md:115]: Yes, this is very confusing and I went back and forth on it.

My take is that the task is installing a compatible runtime. That's why I changed the title. This is the same thing you two are discussing.

**Review by elinor-fung — APPROVED** [review]

**Review by vitek-karas — APPROVED** [review]

**Pull request merged**